analyses/molecular-subtyping-MB/05-subtype-mb-shh.R

```diff
@@ -0,0 +1,389 @@
+                                 dkfz_v12_methylation_subclass_score_mean >= 0.7 & dkfz_v12_methylation_subclass_collapsed == "MB_SHH_4" ~ "SHH_delta",
+                                 TRUE ~ NA_character_))
+
+mb_subtypes <- mb_shh %>%
+  select(match_id, molecular_subtype, molecular_subtype_methyl) %>%
+  unique()
+
+final_subtypes <- mb_subtypes %>%
+  left_join(hist_mb_shh) %>%
+  left_join(mb_shh_data_collapsed) %>%
+  left_join(germline) %>%
+  left_join(expr_df) %>%
+  left_join(tp53_df) %>%
+  unique() %>%
+  # clean up subclass and scores
+  mutate(dkfz_v12_methylation_subclass_score_mean = case_when(dkfz_v12_methylation_subclass_score_mean < 0.7 & 
+                                                                !grepl("SHH", dkfz_v12_methylation_subclass_collapsed) ~ NA_integer_,
+                                                              TRUE ~ dkfz_v12_methylation_subclass_score_mean),
+         dkfz_v12_methylation_subclass_collapsed = case_when((dkfz_v12_methylation_subclass_score_mean < 0.7 | is.na(dkfz_v12_methylation_subclass_score_mean)) & 
+                                                               !grepl("SHH", dkfz_v12_methylation_subclass_collapsed) ~ NA_character_,
+                                                             TRUE ~ dkfz_v12_methylation_subclass_collapsed)) %>%
+  mutate(final_shh_subtype = case_when(
+    # For SHH alpha: >3 years AND (MYCN OR GLI2 amplification OR TPM>=2 OR chr17p loss OR 9p gain with lower conf methyl)
+    SHH_subtype == "SHH_alpha" | (age_at_diagnosis_years >= 2 &
+                                    is.na(SHH_subtype) &
+                                    (grepl("amplification", consensus_CN_MYCN) |
+                                       grepl("amplification", consensus_CN_GLI2) |
+                                       grepl("amplification", consensus_CN_CCND2) |
+                                       !is.na(ELP1_germline_plp) |
+                                       !is.na(TP53_germline_plp) |
+                                       MYCN_TPM_zscore >= 2 |
+                                       GLI2_TPM_zscore >= 2 |
+                                       CCND2_TPM_zscore >= 2 |
+                                       TP53_TPM_zscore < -2 |
+                                       tp53_hotspot > 0 | 
+                                       `17p_loss` == 1 |
+                                       (grepl("1", `9p_gain`) & dkfz_v12_methylation_subclass_collapsed == "MB_SHH_3" & 
+                                          dkfz_v12_methylation_subclass_score_mean >= 0.5)
+                                     )) ~ "SHH alpha",
+    
+    # For SHH delta: (TERT promoter AND DDX3X mutations) OR (>5 years AND (TERT OR DDX3X mutations))
+    SHH_subtype == "SHH_delta" |
+      (age_at_diagnosis_years >= 10 & (U1_mutation == 1 | !is.na(TERT_csq) | !is.na(DDX3X_csq))) ~ "SHH delta",
```

Comment:
```suggestion
    # For SHH delta: (>=10 years AND (TERT OR DDX3X OR SMO mutations OR 14q-))
    SHH_subtype == "SHH_delta" |
      (age_at_diagnosis_years >= 10 & (U1_mutation == 1 | !is.na(TERT_csq) | !is.na(DDX3X_csq) | !is.na(SMO_csq) | `14q_loss` == 1)) ~ "SHH delta",
```
think we need to add SMO and 14q loss here